cmd/kubectl-nfd/subcmd/dryrun.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subcmd
+
+import (
+	"github.com/spf13/cobra"
+	kubectlnfd "sigs.k8s.io/node-feature-discovery/pkg/kubectl-nfd"
+)
+
+var dryrunCmd = &cobra.Command{
+	Use:   "dryrun",
+	Short: "Process a NodeFeatureRule file against a NodeFeature file",
+	Long:  `Process a NodeFeatureRule file against a local NodeFeature file to dry run the rule against a node before applying it to a cluster`,
+	Run: func(cmd *cobra.Command, args []string) {
+		err := kubectlnfd.DryRun(nodefeaturerule, nodefeature)
+		if len(err) > 0 {
+			for _, e := range err {
+				cmd.PrintErrln(e)
```

Comment:
I think if validation fails it should also return non zero exit code, so it can be used in CI etc

Reply:
good idea, I was not thinking in CI, just manual